database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -125,6 +125,16 @@ trait WorkflowMetadataSummaryEntryComponent {
     } yield summaryEntry
   }
 
+  private def fetchAllWorkflowsToDeleteThatEndedOnOrBeforeThresholdTimestamp(workflowStatuses: List[String],
+                                                                              workflowEndTimestampThreshold: Timestamp): Query[WorkflowMetadataSummaryEntries, WorkflowMetadataSummaryEntry, Seq] = {
+    for {
+      summaryEntry <- workflowMetadataSummaryEntries
+      if summaryEntry.workflowStatus.inSet(workflowStatuses)
```

Comment:
I think when looking for workflows to delete, we might not need to check the workflow status. We only need to check for archive status and end time just like [workflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp](https://github.com/broadinstitute/cromwell/blob/187c49defd1ef3c2df4133e6e9b9c92edfce79b4/database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala#L74-L81)